File: recursion/go/Recursion6.go
package go
package main

import "fmt"

// first n natural number
func printFirstNOddNaturalNumberReverse(number int) {
	// negative number
	if number < 1 {
		return
	}

	// base case
	if number == 1 {
		fmt.Print(number, ", ")
		return
	}

	// start with Odd 
if(number % 2 == 0 ){
	 number = number - 1
}
	// recursive call
	fmt.Print(number, ", ")
	printFirstNOddNaturalNumberReverse(number - 2)

}
func main() {
	var number int
	fmt.Scan(&number)
	printFirstNOddNaturalNumberReverse(number)
}

File: recursion/java/Recursion6.java
import java.util.Scanner;

public class Recursion6 {

    public static void printFirstNOddNaturalNumberReverse(int n) {
        // negative number
        if (n < 1) {

            return;
        }

        if (n == 1) {
            System.out.print(n + ", ");
            return;
        }
        // make Odd
        if (n % 2 == 0) {
            n = n - 1;
        }

        System.out.print(n + ", ");
        printFirstNOddNaturalNumberReverse(n - 2);

    }

    public static Scanner scan = new Scanner(System.in);

    public static void main(String... args) {
        int n = scan.nextInt();
        printFirstNOddNaturalNumberReverse(n);
    }
}